### PR TITLE
Update paths and add maven build system

### PR DESCRIPTION
**Pull request by rleigh-codelibre**
The aim of this PR is to take the result of `git filter-branch` in ome-common-java, which is currently not buildable, and to make it build and test with maven, and also do the same driven by travis.  This will get the repository functional and testable so that this basic infrastructure will be in place for future PRs.

- Use standard maven path layout
- Drop ant build
- Add travis support for maven build only
- Make maven build work again
- Update licence and readme

NB.  maven builds needs further work either in this PR or as a followup.  The `pom.xml` is an aggregate of the logic in `components/formats-common` and the toplevel.  Some of the logic is redundant, and some is likely outdated and unnecessary.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,577 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0
+    http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>org.openmicroscopy</groupId>
+  <artifactId>ome-common</artifactId>
+  <version>6.0.0-SNAPSHOT</version>
```
Comment (sbesson) [inline pom.xml]: One question here is whether we should not be conservative and target this as `5.3.0-SNAPSHOT`. As far as I know, our first target is a Maven release with no API change and for consumers the major change is the usage of a different `groupId/artifactId`  (and only if they are declaring it directly in their POM).

Comment (rleigh-codelibre) [inline pom.xml]: This would certainly make sense.

File: .gitignore [hunk 2]
```diff
@@ -1,30 +1,2 @@
...
-Missing OME-XML Service Tests
-Missing POI Service Tests
-OME ReaderWriter Service Tests
-OMEXMLTests
-artifacts
-build
-components/*/utils/*.class
-emailable-report.html
-index.html
 target
```
Comment (sbesson) [inline .gitignore]: Is target actually used at all, it might be another legacy of the Ant system.

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,479 @@
...
+
+  <scm>
+    <connection>scm:git:https://github.com/ome/ome-common-java</connection>
+    <developerConnection>scm:git:git@github.com:ome/ome-common-java</developerConnection>
+    <tag>HEAD</tag>
+    <url>http://github.com/ome/ome-common-java</url>
+  </scm>
+
+  <repositories>
+    <repository>
```
Comment (sbesson) [inline pom.xml]: We might want to remove the OME artifactory links here as nothing should depend on them?

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,479 @@
...
+      <url>http://artifacts.openmicroscopy.org/artifactory/ome.staging</url>
+    </repository>
+    <snapshotRepository>
+      <id>ome.snapshots</id>
+      <name>OME Snapshots Repository</name>
+      <url>http://artifacts.openmicroscopy.org/artifactory/ome.snapshots</url>
+    </snapshotRepository>
+  </distributionManagement>
+
+  <developers>
```
Comment (sbesson) [inline pom.xml]: The developers list might require a review in a follow-up PR.

File: src/test/resources/logback.xml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (sbesson) [inline src/test/resources/logback.xml]: :+1: for restoring this in the correct place as logback initialization is required for some of the `DebugToolsTest` unit tests

File: .gitignore (hunk 2 above)
Comment (sbesson) [inline .gitignore]: Understood, ignore my comment then.

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,479 @@
...
+        <version>2.8.2</version>
+      </plugin>
+
+      <plugin>
+        <artifactId>maven-install-plugin</artifactId>
+        <version>2.5.2</version>
+      </plugin>
+
+      <plugin>
+        <artifactId>maven-jar-plugin</artifactId>
```
Comment (sbesson) [inline pom.xml]: Tested the integration of this component into Bio-Formats on top of https://github.com/openmicroscopy/bioformats/pull/2608. Looking at the difference between the content of `formats-common-5.2.3.jar` and `ome-common-5.3.0-SNAPSHOT.jar`, except for `META-INF`, the only missing file is  `service.properties` which is currently required for using the services.

Comment (rleigh-codelibre) [inline pom.xml]: This is now added.

**Comment by rleigh-codelibre** [comment]
This PR can be tested functionally using https://github.com/rleigh-codelibre/bioformats/tree/common-split

Run `mvn install` for this repository, then check out this branch in bioformats.git and then run `ant` or `mvn` and it will use the ome-common jar for building and testing.

**Comment by sbesson** [comment]
Thanks for the instructions. Using the HEAD of this PR and building successfully `ome-common-java` and `bioformats` with the branch above worked as expected (build + unit tests using Maven). Merging.

**Pull request merged**